### ISSUE TITLE
VS Code doesn't appear a content of the object in debug mode

### ISSUE THREAD

**Issue by isvforall** [Microsoft/vscode#6483]
- VSCode Version: 1.1.1
- OS Version: Mac OS 10.11.4

Steps to Reproduce:

1. Debug objects that comes in ES2016 standard, e.g. `Set` or `Map` 

VS Code doesn't appear a content of the object in debug mode for `Set` or `Map` (see screenshot)

![vscode](https://cloud.githubusercontent.com/assets/706913/15364118/58122cba-1d2c-11e6-978f-0cd652dd396a.png)

**Comment by weinand** [comment]
@isvforall your snippet works fine for me in VS Code 1.1.1 with node 5.11.0:

![2016-05-18 22-03-36](https://cloud.githubusercontent.com/assets/1898161/15373177/8d5c25ca-1d44-11e6-98d2-b6dc84878a57.png)


(BTW, the Map is empty, so it cannot show any content)

What version of node are you using?

**Comment by isvforall** [comment]
@weinand I think `Map` should show `__proto__ : Map {}` like `__proto__ : Object {}` for empty object. My node version is 6.0.0

**Comment by weinand** [comment]
@isvforall the node debugger protocol does not return __proto__ for Set and Map.

**Comment by isvforall** [comment]
I've just tested with node v5.11.0, `Set` shows own content, but it doesn't work for node v6.0.0

**Comment by weinand** [comment]
@isvforall yes, VSCode 1.1.1 does not yet support node 6.0.

**Comment by isvforall** [comment]
@weinand 
> the node debugger protocol does not return proto for Set and Map.

What it means? I was thinking that node.js and Chrome use the same v8 debugger protocol.

I was debugging that snippet in Chrome, and Chrome shows a `__proto__` for `Map`:

![chrome](https://cloud.githubusercontent.com/assets/706913/15377222/fbd19eec-1d6a-11e6-8a63-381320cfd95e.png)
 
Also I have created a node.js project in WebStorm, and it is also show me a `__proto__` for `Map`  in debug mode:

![idea](https://cloud.githubusercontent.com/assets/706913/15377238/46a264d8-1d6b-11e6-93d8-ab422fb1ff7d.png)

**Comment by weinand** [comment]
@isvforall no, neither Chrome Dev Tools nor WebStorm use the v8 debugger protocol. The Chrome Dev Tools use the Chrome Debugging Protocol which is not (yet) supported by node and WebStorm uses code injection to implement its own debug protocol.
Node-inspector uses the v8 debugger protocol. Try it to compare it with VS Code.

VS Code uses the v8 debugger protocol but it has to work around the limitations by a combination of code injection and use of eval. Supporting `__proto__` on Maps and Sets is possible but it has not a high priority on my todo list. But feel free to submit a PR.
